tests/components/hyperion/test_switch.py

```diff
@@ -136,3 +149,51 @@ async def test_switch_has_correct_entities(hass: HomeAssistantType) -> None:
         )
         entity_state = hass.states.get(entity_id)
         assert entity_state, f"Couldn't find entity: {entity_id}"
+
+
+async def test_device_info(hass: HomeAssistantType) -> None:
+    """Verify device information includes expected details."""
+    client = create_mock_client()
+    client.components = TEST_COMPONENTS
+
+    with patch(
+        "homeassistant.components.hyperion.switch.HyperionComponentSwitch.entity_registry_enabled_default"
+    ) as enabled_by_default_mock:
+        enabled_by_default_mock.return_value = True
```

Comment:
We should not patch the implementation in tests.
Instead, reserve the switch entity in the entity registry, like this:
https://github.com/home-assistant/core/blob/155322584d2bc2203b0e658ae075a5e930af73b3/tests/components/tasmota/test_sensor.py#L232-L239
Please update the other tests too in the same way, both for light and switch.

Please also add tests which make sure that entities can be enabled, like this:
https://github.com/home-assistant/core/blob/155322584d2bc2203b0e658ae075a5e930af73b3/tests/components/tasmota/test_sensor.py#L547-L559